### ISSUE TITLE
SOLR/GetOccurrences did not provide occurrences from 2024-11-28T02:00 - 2024-11-29T15:00 for 3 National Channels:    "StationIDs" :["INSPIRATION", "REELZ", "VICE"],

### ISSUE THREAD

**Issue by sergi0aranda** [iQMedia/tracker#67]
@jmylet @hollinger - Please research why SOLR/GetOccurrences is not providing any hits for these 3 channels:    "StationIDs" :["INSPIRATION", "REELZ", "VICE"] during the following period: 11/28 02:00 UTC – 11/29 15:00 UTC.

Here's the request:

curl --location 'https://apiservices.iqmediacorp.com/isvc/v2/GetOccurrences' \
--header 'Content-Type: application/json' \
--header 'Accept: application/json' \
--header 'Authorization: Bearer "{{SessionID}}"' \
--header 'Cookie: Cookie_1=value; UqZBpD3n=v1QdMkgw__iqT' \
--data '{
  
    "AirDateStartUtc": "2024-11-28T02:00",
    "AirDateEndUtc": "2024-11-29T15:00",
    "ExcludeDeleted": true,
    "IncludeDetailedUpdates": false,
    "IncludeCreativeMetadata": true,
    "SortFields": [
        "startdateutc asc"],

   "StationIDs" :["INSPIRATION", "REELZ", "VICE"], 
    
   
    "CursorMark": "*",
    
    "Rows": 500,
    "SessionID": "{{SessionID}}"
}'

Here's the response:

{
    "TotalResults": 0,
    "NextCursorMark": "*",
    "Message": "Success",
    "Status": 0
}

Attached you can see **3,743** full hits during this same period in Teletrax.

Please advise, thanks

[VICE_REELZ_INSPIRATIONAL_2024_12_03_00_35_46.xlsx](https://github.com/user-attachments/files/17985507/VICE_REELZ_INSPIRATIONAL_2024_12_03_00_35_46.xlsx)

**Comment by jmylet** [comment]
We already have documents in solr for REELZ going all the way back to 2018-12-15, so this channel must have had a previous IQ_Station_History record with a StartDate going back at least that far for these documents to exist so we can probably reset REELZ back to around the end of 2018 or the beginning of 2019. 
As for VICE and INSPIRATION, the earliest available solr documents start after `2024-11-29T15:00` for both channels so they probably weren't being ingested into solr prior to these latest IQ_Station_History records being created and so I don't really have any thoughts as to when either of those StartDates should be set.

**Comment by sergi0aranda** [comment]
Thanks @jmylet . So ok on REELZ to reset back to 2019-01-01.
For Vice and Inspiration, I definitely have proof that we had those channels before, but like you can see in the attached screenshot, someone deleted them and created them again with new Station IDs.  Here you can see some of the occurrences we had in some of the files (that we pushed to Comscore) in the first months of the year. The last column you see is the occurrence count for that particular day. I would say start them in 2024-01-01, but I want to understand what happens if we do that? Do we push the occurrences again into SOLR but with the new station ID?
![Screenshot 2024-12-04 at 1 58 31 PM](https://github.com/user-attachments/assets/1e8479f9-1249-4b16-93c8-f82286c8a6d2)

**Comment by jmylet** [comment]
If we were to push back the start dates of VICE and INSPIRATION to 2024-01-01 then we would need to replace all occurrences with the old VICEL and INSP station IDs that occurred on or after 2024-01-01 because the occurrence solr collection cannot contain any documents with the same client ID, asset ID, and client hit ID (these values are joined together which then will function as a unique key to identify each document). If we replace all of these documents then it would break any previously valid query for occurrences on the VICEL and INSP channels on or after 2024-01-01 (or whatever we set it to) by returning 0 results, which I'm guessing may concern some of our clients to suddenly not see certain data anymore, this would also essentially invalidate those files we had pushed to Comscore for early 2024. I'm also not sure if replacing occurrences for the VICEL and INSP channels on or after 2024-01-01 is really even conceptually valid unless these stations were always supposed to be VICE and INSPIRATION, in which case we would probably need to replace ALL occurrences in solr for the VICEL and INSP channels going back to the earliest documents we have available, which would be 2018-12-15. Another potentially problem with updating historical occurrence documents from before the new VICE and INSPIRATION station records were created would be that this will also change the associated IQCCKeys to use the new station IDs (e.g. into something like VICE_20181215_1700 & INSPIRATION_201821215_1700), which would almost certainly be invalid/incorrect because those cc files never existed within our system in the first place. Because of these issues I would say that we probably shouldn't move back the start date for either of the new VICE and INSPIRATION station history records, otherwise we could create a larger mess by forcing ingestion to function in way it was designed not to.
